psi4/src/export_mints.cc

```diff
@@ -1304,7 +1304,7 @@ void export_mints(py::module& m) {
         .def("shell_significant", compute_shell_significant(&TwoBodyAOInt::shell_significant),
              "Determines if the P,Q,R,S shell combination is significant");
 
-    py::class_<Libint2ERI, std::shared_ptr<Libint2ERI>>(m, "ERI", pyTwoBodyAOInt,
+    py::class_<Libint2ERI, std::unique_ptr<Libint2ERI>>(m, "ERI", pyTwoBodyAOInt,
```

Comment:
@loriab  @JonathonMisiewicz  After many hours staring at valgrind I figured out the issue was that the PyBind11 holder type for the `eri` needed to be a unique ptr to avoid  default deletion of the `ERI` prior to calling the `screen_impl_` lambda function. Do we anticipate this being an issue with the other `2BodyAOInts`? Are they well covered by tests?

Reply:
Can you go into more detail about what the original problem was and why shared-vs-unique matters? It isn't clear to me what default deletion of `ERI` means, and I can't see which test was failing.

Reply:
Originally the tests in `test_erisieve.py` were failing due to a segfault when calling any of the methods associated with the `ERI` TwoBodyAOInt object, called through the `_screen_impl` lambda.

 This is because the `unique_ptr` holding the C++ TwoBodyAOInt goes is allowed to expire because the `sharedptr` that holds the python class object has no reference counts.  

Making it a `unique_ptr` seems to fix the reference counting on the python object but I'm not exactly sure why.

Reply:
Perhaps it is because the lambda doesn't extend the lifetime, but I'm fairly sure the issue is with the PyBind11 layer. Is anyone here in particular a Pybind expert?